juris-ca.csl

```diff
@@ -14,14 +14,47 @@
     <summary>Juris-M style module for Canada</summary>
     <updated>2020-10-19T15:45:45.469Z</updated>
     <rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>
-    <law-module types="legal_case legislation bill"/>
+    <law-module types="legal_case legislation bill hearing report regulation gazette"/>
   </info>
   <macro name="juris-pretitle">
     <!-- Fragment to precede title (rare) -->
     <choose>
       <if type="bill">
         <number variable="number" prefix="Bill "/>
       </if>
+      <else-if type="hearing">
+        <group delimiter=", ">
+          <group delimiter="">
+            <text term="open-quote"/>
+            <text variable="event"/>
+            <text term="close-quote"/>
+          </group>
+          <text variable="references"/>
+          <names variable="authority"/>
+          <names variable="committee"/>
+        </group>
+      </else-if>
+      <else-if type="report">
+        <names variable="authority"/>
+      </else-if>
+      <else-if>
+        <conditions match="all">
+          <condition match="all" variable="authority"/>
+          <condition match="all" type="regulation"/>
+        </conditions>
+        <group delimiter=", ">
+          <names variable="authority"/>
+          <group delimiter=" ">
+            <text variable="genre"/>
+            <choose>
```

Comment:
So this is a particular case. In this instance, the regulation item is being used to render a municipal bylaw (since there isn't a way to specify a bylaw item type, this was the closest item type I could think of).

In Canada, bylaws are formatted in one of two ways, depending on whether they have been revised into a municipal code, or if they have been adopted after the last code revision. 

Revised bylaws are rendered as "[City], revised by-law [chapter number], [title], ([date]), [pinpoint]".
Unrevised bylaws are rendered as "[City], by-law No [by-law number], [title], [pinpoint].

So, as you can see, revised bylaws lack the date, and also don't have a "No" term, since the by-law number is replaced by a chapter number. Unrevised bylaws have both the date in which it was adopted, as well as a "No" term preceding the by-law number.

This assumes that the "genre" variable is used to indicate either "by-law" or "revised by-law". In either case, it should always be present. I should probably add it as a condition in this case. 

The "volume" variable is used as either the chapter number for a revised bylaw, or the by-law number for an unrevised by-law. 

So what this code does is checks whether or not the "issued" variable is supplied. If the "issued" variable is present, it is assumed that the item is an unrevised bylaw, and this macro should return [genre] No [volume]. 
If there is no "issued" variable, then the item is assumed to be an unrevised by-law, and the macro should return [genre] [volume].

In either case, there should always be a "genre" and a "volume" variable supplied. However, writing this comment has made me realize that I should add the presence of these two variables as a condition to render this macro, since they have to be present for this to work correctly.

Reply:
No, I don't think you should use a conditional rendering here. Just do this:

```
<group delimiter=" ">
  <text variable="genre"/>
  <group delimiter=" ">
    <text term="issue"/>
    <text variable="volume"/>
  </group>
</group>
```

That will ensure that bare labels are never rendered. If data exists here with genre, no year, but not volume number, it seems more correct to show the genre than to omit it.

Reply:
The problem with what you're suggesting is that the condition rendering of the "issue" term is lost. 

"Issue" should only render when the item has an "issued" value, as that indicates it is an unrevised by-law. 

"issue" shouldn't render if there is no "issued" date value because that would indicate it is a revised by-law.

So I've pushed a modification that does both. It separates "issue" and "volume" into their own group, but only renders the "issue" term if the "issued" date variable is present.

Reply:
But just to clarify = there should never be a case where this macro is run and the "volume" variable is empty. If that happens, there is a fundamental error in the item.